### ISSUE TITLE
leveldb shared library doesn't list snappy as a dependency.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmithdemos/leveldb#94]
```
What steps will reproduce the problem?
1. from a fresh clone, `make`
2. `ldd libleveldb.so.1`

What is the expected output? What do you see instead?
snappy should be listed, but it is not.

What version of the product are you using? On what operating system?
i'm using leveldb 1.5 on ubuntu 12.04

Please provide any additional information below.
the problem is that the '-lsnappy' option (from the generated build_config.mk's 
PLATFORM_LDFLAGS) is being provided before the list of source files. if it is 
provided after, then the dependency is properly listed in the generated so file.
This can be fixed by moving $(SOURCES) to be a sooner option in the $(SHARED3) 
target.
```

---

Original issue reported on code.google.com by `jtolds` on 5 Jun 2012 at 6:05

**Comment by GoogleCodeExporter** [comment]
```
This is critical to use shared leveldb with python ctypes (see 
http://code.google.com/p/leveldb-py/) so, please fix this. I attached a patch.
```

---

Original comment by `carlopi...@gmail.com` on 31 Aug 2012 at 2:07

Attachments:
- [leveldb-shared-fix.patch](https://storage.googleapis.com/google-code-attachments/leveldb/issue-94/comment-1/leveldb-shared-fix.patch)

**Comment by GoogleCodeExporter** [comment]
```
For anyone waiting on this issue, I have a fork of LevelDB that both fixes this 
and adds Debian packaging: http://code.google.com/r/jtolds-leveldb/
```

---

Original comment by `jtolds` on 31 Aug 2012 at 4:05

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.6.0.

```

---

Original comment by `san...@google.com` on 12 Oct 2012 at 7:58
- Changed state: **Fixed**
